src/network/networkd-sysctl.c

```diff
@@ -315,11 +376,44 @@ static int link_set_ipv4_promote_secondaries(Link *link) {
         return sysctl_write_ip_property_boolean(AF_INET, link->ifname, "promote_secondaries", true, &link->sysctl_shadow);
 }
 
+static void link_watch_sysctl(Link *link) {
+        sd_event_source **source;
+        int r;
+
+        assert(link);
+
+        source = link->sysctl_event_sources;
+
+        FOREACH_STRING(p,
+                "/proc/sys/net/ipv4/conf",
+                "/proc/sys/net/ipv6/conf") {
+                _cleanup_free_ char *path = path_join(p, link->ifname);
+
+                if (!path)
+                        return;
+
+                /* We're reconfiguring the link, an inotify watcher is already present for this path */
```

Comment:
If we're reconfiguring the link ... ?

Reply:
Is `link_set_sysctl` called only once in the lifetime of a `Link` object?